packages/tracking_repository/lib/src/tracking_repository.dart

```diff
@@ -21,8 +21,47 @@ class TrackingRepository {
   void track(TrackEvent event) {
     _firebase.logEvent(name: event.name);    
     dev.log(event.name);
-    // firebase
-    // GA
-    // tracker?.track...
+  }
+
+  /// Updates tracking services when the user has approved a swap
+  void trackTradeApproval(
```

Comment:
all these Events should be created on the BL side. The should not have any knowledge on what are the business requirements of "building events" it just forwards the ones received to different services.